### DISCUSSION TITLE
How do I know if an element is updated and not being updated by the ResaveElements task?

### DISCUSSION THREAD

**Discussion by BenParizek** [craftcms/cms#9669] [Q&A] [answered]
Is there a reliable way to determine if an Element is being updated without matching a mass resave event? While checking if an Element is not a draft and not a revision can help identify this scenario, this is also true for the ResaveElements task. In my tests, the element properties during a ResaveElements task are:

| isCanonical | isDraft | isProvisionalDraft | isRevision | scenario |
| -- | -- | -- | --| -- |
| TRUE | FALSE | FALSE | FALSE | essentials |

Is there any way to ensure we exclude the ResaveElements scenario?

**Comment by michaelrog** [comment, marked as answer]
Check out [`$element->resaving`](https://docs.craftcms.com/api/v3/craft-base-elementtrait.html#resaving).
